metrics-operator/controllers/common/providers/dynatrace/client/config.go

```diff
@@ -27,18 +32,23 @@ func WithScopes(scopes []OAuthScope) APIConfigOption {
 }
 
 // NewAPIConfig returns a new apiConfig that can be used for initializing a DTAPIClient with the NewAPIClient function
-func NewAPIConfig(serverURL string, secret string, opts ...APIConfigOption) (*apiConfig, error) {
-	if err := validateOAuthSecret(secret); err != nil {
+func NewAPIConfig(serverURL string, secret []byte, opts ...APIConfigOption) (*apiConfig, error) {
+	var secValue secretValues
+	if err := json.Unmarshal(secret, &secValue); err != nil {
+		return nil, err
+	}
+
+	if err := validateOAuthSecret(secValue.Token); err != nil {
```

Comment:
let's extend the validation function also for validating the URL

Reply:
hey @odubajDT, what needs to be more configured here.....is it the presence of the token in the authurl

Reply:
for now I would check that the url is a valid url and not a random string, for example you can use https://pkg.go.dev/net/url#ParseRequestURI to check this

Reply:
thanks